remote-dependency-templates/README.md

```diff
@@ -3,9 +3,73 @@
   - Populate required `template-config` fields (defined in the README for each type)
 - List any dependency-specific service data under `dependency-data.<id>` in a service's `service.yml`
 
-## Terraform variables //TODO
+# Building dependency templates
+- Create a directory with a unique name. This will be the dependency's `type`
 
-## Notes
-The following are fields that exosphere passes to every dependency:
-- `app-name`: Name of the application as defined in `application.yml`
+## requirements.yml
+`requirements.yml` contains a single field: `required-fields`, which is a list of strings that define the fields that a user is required to populate
+under the `template-config` section of a dependency configuration.
 
+Example:
+```yml
+# requirements.yml
+
+required-fields:
+  - version
+```
+```yml
+# application.yml
+
+<dependency-id>:
+  type: <type>
+  template-config:
+    version: 0.0.1
+```
+
+## dependency.tf
+Add a terraform file, `dependency.tf`, which contains the public interfacing terraform modules for a dependency. The `dependency.tf` file should be built
```

Comment:
Since this is under the section `dependency.tf`, I don't think we need to repeat the name everywhere. Instead it could be something like. Some potential changes:
* `Add a terraform file, `dependency.tf`, which` -> `A terraform file which`
*  `The `dependency.tf` file should` -> The file should